public/app/features/dashboard/state/DashboardModel.ts

```diff
@@ -365,11 +370,30 @@ export class DashboardModel {
   exitViewPanel(panel: PanelModel) {
     this.panelInView = undefined;
     panel.setIsViewing(false);
+    this.refreshIfChangeAffectsAllPanels();
   }
 
   exitPanelEditor() {
     this.panelInEdit!.destroy();
     this.panelInEdit = undefined;
+    this.refreshIfChangeAffectsAllPanels();
+  }
+
+  changeAffectsAllPanels() {
+    if (!this.panelInEdit && !this.panelInView) {
+      return;
+    }
+
+    this.hasChangesThatAffectsAllPanels = true;
```

Comment:
maybe more intuitive to write this as:

```ts
if (this.panelInEdit || this.panelInView) {
   this.hasChangesThatAffectsAllPanels = true;
}
```
That way you do not have to invert the booleans in your to understand when this is set to true :)

or just  this.hasChangesThatAffectsAllPanels = this.panelInEdit || this.panelInView;

Reply:
I was thinking about that too, thanks!

Reply:
Fixed!